src/js/components/TextInput/TextInput.js

```diff
@@ -444,12 +468,31 @@ const TextInput = forwardRef(
 
     const textInputIcon = useSizedIcon(icon, rest.size, theme);
 
+    const CopyButton = (
+      <Tip dropProps={{ align: { bottom: 'top' } }} content={tip}>
+        <Button
+          onClick={onClickCopy}
+          plain
+          icon={<CopyIcon />}
+          margin={reverse ? { left: 'small' } : { right: 'small' }}
+          onBlur={onBlurCopy}
+          onMouseOut={onBlurCopy}
+          aria-label={`${readOnlyCopyPrompt} ${value}`}
+        />
+      </Tip>
+    );
+
     return (
-      <StyledTextInputContainer plain={plain}>
+      <StyledTextInputContainer
+        readOnlyCopy={readOnlyCopy}
+        plain={plain}
+        border={!plain}
```

Comment:
what is this? why is the `border` needed?

Reply:
This container goes around the Button and the input. I have a border on it so that the border doesn't just go around the input and the button looks like it is outside of the container.

<img width="382" alt="Screen Shot 2024-02-16 at 11 13 11 AM" src="https://github.com/grommet/grommet/assets/54560994/72cc9b9a-1711-4ff4-a781-6e0b2840a5ce">

Reply:
got it thank you for clarifying I was a little lost